package compilation fails on R 3.6.3.1 on Cygwin

Hi,
I've sent a similar request to the Cygwin mailing list in case this is a problem of the R port.
uname -a
CYGWIN_NT_10.0 PCXYZ 3.1.4(0.340/5/3) 2020-02-19 08:49 x86_64 Cygwin
R --version
R version 3.6.3 (2020-02-29) -- "Holding the Windsock"

$ R
[...]
install.packages("processx")
[...]
* installing *source* package ‘processx’ ...
** Paket ‘processx’ erfolgreich entpackt und MD5 Summen überprüft
** using staged installation
** libs
gcc -ggdb -O2 -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=4 -std=gnu99 -fdebug-prefix-map=/cygdrive/d/cyg_pub/devel/R/R-3.6.3-1.x86_64/build=/usr/src/debug/R-3.6.3-1 -fdebug-prefix-map=/cygdrive/d/cyg_pub/devel/R/R-3.6.3-1.x86_64/src/R-3.6.3=/usr/src/debug/R-3.6.3-1  -Wall tools/px.c -o tools/px
gcc -ggdb -O2 -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=4 -std=gnu99 -fdebug-prefix-map=/cygdrive/d/cyg_pub/devel/R/R-3.6.3-1.x86_64/build=/usr/src/debug/R-3.6.3-1 -fdebug-prefix-map=/cygdrive/d/cyg_pub/devel/R/R-3.6.3-1.x86_64/src/R-3.6.3=/usr/src/debug/R-3.6.3-1  supervisor/supervisor.c supervisor/utils.c \
      -o supervisor/supervisor
gcc -I"/usr/lib/R/include" -DNDEBUG       -ggdb -O2 -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=4 -std=gnu99 -fdebug-prefix-map=/cygdrive/d/cyg_pub/devel/R/R-3.6.3-1.x86_64/build=/usr/src/debug/R-3.6.3-1 -fdebug-prefix-map=/cygdrive/d/cyg_pub/devel/R/R-3.6.3-1.x86_64/src/R-3.6.3=/usr/src/debug/R-3.6.3-1  -c base64.c -o base64.o
gcc -I"/usr/lib/R/include" -DNDEBUG       -ggdb -O2 -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=4 -std=gnu99 -fdebug-prefix-map=/cygdrive/d/cyg_pub/devel/R/R-3.6.3-1.x86_64/build=/usr/src/debug/R-3.6.3-1 -fdebug-prefix-map=/cygdrive/d/cyg_pub/devel/R/R-3.6.3-1.x86_64/src/R-3.6.3=/usr/src/debug/R-3.6.3-1  -c client.c -o client.o
gcc -I"/usr/lib/R/include" -DNDEBUG       -ggdb -O2 -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=4 -std=gnu99 -fdebug-prefix-map=/cygdrive/d/cyg_pub/devel/R/R-3.6.3-1.x86_64/build=/usr/src/debug/R-3.6.3-1 -fdebug-prefix-map=/cygdrive/d/cyg_pub/devel/R/R-3.6.3-1.x86_64/src/R-3.6.3=/usr/src/debug/R-3.6.3-1  -c errors.c -o errors.o
gcc -shared -L"/usr/lib/R/lib"  -o client.dll base64.o client.o errors.o
/usr/lib/gcc/x86_64-pc-cygwin/9.3.0/../../../../x86_64-pc-cygwin/bin/ld: base64.o: in function `processx_base64_encode':
/tmp/RtmpyuAKvT/R.INSTALL2dfd356fee1/processx/src/base64.c:79: undefined reference to `RAW'
/tmp/RtmpyuAKvT/R.INSTALL2dfd356fee1/processx/src/base64.c:79:(.text+0x14): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `RAW'
/usr/lib/gcc/x86_64-pc-cygwin/9.3.0/../../../../x86_64-pc-cygwin/bin/ld: /tmp/RtmpyuAKvT/R.INSTALL2dfd356fee1/processx/src/base64.c:80: undefined reference to `LENGTH'
/tmp/RtmpyuAKvT/R.INSTALL2dfd356fee1/processx/src/base64.c:80:(.text+0x1f): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `LENGTH'
/usr/lib/gcc/x86_64-pc-cygwin/9.3.0/../../../../x86_64-pc-cygwin/bin/ld: /tmp/RtmpyuAKvT/R.INSTALL2dfd356fee1/processx/src/base64.c:82: undefined reference to `Rf_allocVector'
[...and so forth...]

The missing symbols or functions are defined in /usr/lib/R/include/Rinternals.h. That file is present and readable.

Any help is appreciated.
Markus

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -23,7 +23,7 @@ test_script:
   - travis-tool.sh run_tests
 
 on_success:
-  - Rscript -e "options(covr.gcov = 'C:/Rtools/mingw_64/bin/gcov.exe');covr::codecov()"
+  - Rscript -e "covr::codecov()"
 
 on_failure:
   - travis-tool.sh dump_logs

--- a/src/Makevars
+++ b/src/Makevars
@@ -20,7 +20,7 @@ supervisor/supervisor: supervisor/supervisor.c supervisor/utils.c
 CLIENT_OBJECTS = base64.o client.o errors.o
 
 client$(SHLIB_EXT): $(CLIENT_OBJECTS)
-	$(SHLIB_LINK) -o client$(SHLIB_EXT) $(CLIENT_OBJECTS) $(PKG_LIBS)
+	$(SHLIB_LINK) -o client$(SHLIB_EXT) $(CLIENT_OBJECTS) $(PKG_LIBS) $(SHLIB_LIBADD) $(LIBR)
 
 clean:
 	rm -rf $(SHLIB) $(OBJECTS) $(CLIENT_OBJECTS)		\

--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -27,7 +27,7 @@ supervisor/supervisor.exe: supervisor/supervisor.c supervisor/utils.c \
 CLIENT_OBJECTS = base64.o client.o errors.o
 
 client$(SHLIB_EXT): $(CLIENT_OBJECTS)
-	$(SHLIB_LINK) -o client$(SHLIB_EXT) $(CLIENT_OBJECTS) $(LIBR) $(SHLIB_LIBADD) $(PKG_LIBS)
+	$(SHLIB_LINK) -o client$(SHLIB_EXT) $(CLIENT_OBJECTS) $(PKG_LIBS) $(SHLIB_LIBADD) $(LIBR)
 
 clean:
 	rm -rf $(SHLIB) $(OBJECTS) $(CLIENT_OBJECTS)			\

